maintainers/scripts/remove-old-aliases.py

```diff
@@ -69,67 +89,103 @@ def get_date_lists(
         date_sep_line_list,
+        date_too_complex_list,
         date_older_throw_list,
+        date_older_warning_list,
     )
 
 
-def convert_to_throw(date_older_list: list[str]) -> list[tuple[str, str]]:
-    """convert a list of lines to throws"""
-    converted_list = []
-    for line in date_older_list.copy():
+def convert(lines: list[str], convert_to: str) -> list[tuple[str, str]]:
+    """convert a list of lines to either "throws" or "warnings"."""
+    converted_lines = []
+    for line in lines.copy():
         indent: str = " " * (len(line) - len(line.lstrip()))
-        before_equal = ""
-        after_equal = ""
-        try:
-            before_equal, after_equal = (x.strip() for x in line.split("=", maxsplit=2))
-        except ValueError as err:
-            print(err, line, "\n")
-            date_older_list.remove(line)
-            continue
 
-        alias = before_equal
-        alias_unquoted = before_equal.strip('"')
-        replacement = next(x.strip(";:") for x in after_equal.split())
+        if "=" not in line:
+            assert "inherit (" in line
+            before, sep, after = line.partition("inherit (")
+            inside, sep, after = after.partition(")")
+            if not sep:
+                print(f"FAILED ON {line}")
+                continue
+            alias, *_ = after.strip().split(";")[0].split()
+            replacement = f"{inside.strip()}.{alias}"
+
+        else:
+            before_equal = ""
+            after_equal = ""
+            try:
+                before_equal, after_equal = (
+                    x.strip() for x in line.split("=", maxsplit=2)
+                )
+            except ValueError as err:
+                print(err, line, "\n")
+                lines.remove(line)
+                continue
+
+            alias = before_equal
+            replacement = next(x.strip(";:") for x in after_equal.split())
+
+        alias_unquoted = alias.strip('"')
         replacement = replacement.removeprefix("pkgs.")
 
-        converted = (
-            f"{indent}{alias} = throw \"'{alias_unquoted}' has been"
-            f" renamed to/replaced by '{replacement}'\";"
-            f" # Converted to throw {datetime.today().strftime('%Y-%m-%d')}"
-        )
-        converted_list.append((line, converted))
+        if convert_to == "throws":
+            converted = (
+                f"{indent}{alias} = throw \"'{alias_unquoted}' has been"
+                f" renamed to/replaced by '{replacement}'\";"
+                f" # Converted to throw {datetime.today().strftime('%Y-%m-%d')}"
+            )
+            converted_lines.append((line, converted))
+        elif convert_to == "warnings":
+            converted = (
+                f"{indent}{alias} = lib.warnOnInstantiate \"'{alias_unquoted}' has been"
+                f" renamed to/replaced by '{replacement}'\" {replacement};"
+                f" # Converted to warning {datetime.today().strftime('%Y-%m-%d')}"
+            )
```

Comment:
Unfortunately, some aliases point to non‐derivations, and `lib.warnOnInstantiate` will silently produce weird results for those. This can’t be detected without eval. I think that with the timeline we’re hoping for this isn’t a blocker, though – we can manually run a check to ensure that everything being passed to `lib.warnOnInstantiate` is a derivation after conversion.